### PR TITLE
MM-56536 Do not update team members scheme roles if user is guest

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
If a user has the SchemeGuest set in their Team membership, ask the member to be promoted as user before being able to change the scheme to member or admin

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-56536

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by sbishel — APPROVED** [review]
Approving but a quick question. When user is updated to a System User, I assume all channel/team memberships are updated as well.

**Comment by enahum** [comment]
> Approving but a quick question. When user is updated to a System User, I assume all channel/team memberships are updated as well.

@sbishel yes it updates the memberships accordingly

File: server/channels/app/team.go [hunk 1]
```diff
@@ -510,14 +510,18 @@ func (a *App) UpdateTeamMemberSchemeRoles(c request.CTX, teamID string, userID s
...
 
-	member.SchemeAdmin = isSchemeAdmin
-	member.SchemeUser = isSchemeUser
-	member.SchemeGuest = isSchemeGuest
+	if member.SchemeGuest {
+		return nil, model.NewAppError("UpdateTeamMemberSchemeRoles", "api.team.update_team_member_roles.guest.app_error", nil, "", http.StatusBadRequest)
+	}
 
-	if member.SchemeUser && member.SchemeGuest {
+	if isSchemeUser && isSchemeGuest {
```
Comment (jespino) [inline server/channels/app/team.go:517]: What happens if I pass `isSchemeAdmin` and `isSchemeGuest` as true but `isSchemeUser` as false?

Comment (enahum) [reply server/channels/app/team.go:517]: The above if statement should be met and the request rejected with a bad request

Comment (enahum) [reply server/channels/app/team.go:517]: Oh wait not really as is not looking for the flags but the fact that the user was already a guest.

I guess that would be allowed and it probably shouldn't.

I can add that too... thanks @jespino

**Review by jespino — APPROVED** [review]

**Review by esarafianou — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
I'm QA-deferring this PR and will test post merge

**Pull request merged**